infrastructure/constructs/lambda_endpoint.py

```diff
@@ -25,8 +25,8 @@ def __init__(
     ):
         super().__init__(
             scope,
-            f"{env_name}-{construct_id}-function",
-            function_name=f"{env_name}-{construct_id}",
+            f"{construct_id}-function",
```

Comment:
Does this mean the function name will be identical in nonprod and prod?

Reply:
No, the construct ID is the thing which has the "env-" prefix for non-prod environments. So the names should only be different in prod.